src/he/mss.c

```diff
@@ -0,0 +1,196 @@
+/* *
+ * Lightway Core
+ * Copyright (C) 2023 Express VPN International Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "mss.h"
+#include "network.h"
+
+void he_internal_clamp_mss(he_conn_t *conn, uint8_t *packet, size_t *length, size_t capacity) {
```

Comment:
Suggestion: How about we pass the effective_pmtu to this function as an argument, then make the function independent of the `conn` states? i.e making this function a util function which can set mss option on the given packet if it's a TCP SYN.

I think it will make the testing easier. For example, the test code could create a simple tcp socket, connect to 1.1.1.1 then call this function to update mss.

Reply:
I've used scapy to generate packets instead an Idea by @kp-mariappan-ramasamy. This way our unittests wouldn't require some kind of raw socket to work.